File: cdk-from-cfn-testing/expected/sam_nodejs_lambda_arr_transform/golang/SAMNodeJSLambdaArrStack.go
package main

import (
	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	sam "github.com/aws/aws-cdk-go/awscdk/v2/awssam"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type SAMNodeJSLambdaArrStackProps struct {
	cdk.StackProps
}

type SAMNodeJSLambdaArrStack struct {
	cdk.Stack
}

func NewSAMNodeJSLambdaArrStack(scope constructs.Construct, id string, props *SAMNodeJSLambdaArrStackProps) *SAMNodeJSLambdaArrStack {
	var sprops cdk.StackProps
	if props != nil {
		sprops = props.StackProps
	}
	stack := cdk.NewStack(scope, &id, &sprops)

	stack.AddTransform(jsii.String("AWS::Serverless-2016-10-31"))

	sam.NewCfnFunction(
		stack,
		jsii.String("MyFunction"),
		&sam.CfnFunctionProps{
			Runtime: jsii.String("nodejs20.x"),
			Handler: jsii.String("index.handler"),
			InlineCode: jsii.String("exports.handler = async (event) => {\n  console.log(event);\n}\n"),
		},
	)

	return &SAMNodeJSLambdaArrStack{
		Stack: stack,
	}
}

File: cdk-from-cfn-testing/expected/sam_nodejs_lambda_arr_transform/typescript/SAMNodeJSLambdaArrStack.ts
import * as cdk from 'aws-cdk-lib';
import * as sam from 'aws-cdk-lib/aws-sam';

export interface SAMNodeJSLambdaArrStackProps extends cdk.StackProps {
}

export class SAMNodeJSLambdaArrStack extends cdk.Stack {
  public constructor(scope: cdk.App, id: string, props: SAMNodeJSLambdaArrStackProps = {}) {
    super(scope, id, props);

    // Transforms
    this.addTransform('AWS::Serverless-2016-10-31');

    // Resources
    const myFunction = new sam.CfnFunction(this, 'MyFunction', {
      runtime: 'nodejs20.x',
      handler: 'index.handler',
      inlineCode: 'exports.handler = async (event) => {\n  console.log(event);\n}\n',
    });
  }
}
